### PR TITLE
Implemented frienships 

### PR DESCRIPTION
**Pull request by jdmartinez1062**
Implemented friendships functionality along with pending friend requests and friend requests pages.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -9,4 +9,46 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :inverse_friendship, foreign_key: :friend_id, class_name: 'Friendship'
+  has_many :friendships, foreign_key: :user_id, class_name: 'Friendship'
```
Comment (kobiyoyo) [inline app/models/user.rb]: this should be written like this `has_many :friendships`

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -9,4 +11,35 @@ def show
...
+  def my_self?
+    user = find_user
+    return unless current_user == user
+
+    flash[:warnign] = 'You cannot befriend yourself'
+    redirect_to users_path
+  end
+
+  def already_sent?
+    retunr unless (fstatus = current_user.status?(find_user))
```
Comment (kobiyoyo) [inline app/controllers/users_controller.rb]: this should be `return`

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team      
#### Great Job so far, but there are a few more things required. 
### General
- [ ] I can see Posts from other users that are not my friends.
- [ ] No button to  reject a friendship request
- [ ] Anytime i accept one friend request, all other friends are added to my friends list.
- [ ] check comments below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by Ispirett — APPROVED** [review]
# Status Approved  :1st_place_medal: 
Great job @jdmartinez1062  with changes made, you did well. :1st_place_medal:

**Pull request merged**